primefaces/src/main/resources/META-INF/resources/primefaces/message/message.js

```diff
@@ -20,7 +20,8 @@ PrimeFaces.widget.Message = PrimeFaces.widget.BaseWidget.extend({
     init: function(cfg) {
         this._super(cfg);
 
-        var text = this.jq.find('.ui-message-error-detail').text();
+        var summaryOrDetail = this.jq.find('.ui-message-error-summary').length ? "summary" : "detail";
+        var text = this.jq.find('.ui-message-error-' + summaryOrDetail).text();
```

Comment:
TBH i dont like that code as you cant search via IDE for "ui-message-error-detail"